### PR TITLE
Migrate to latest lints

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
+    type_annotate_public_apis: ignore
+    lines_longer_than_80_chars: ignore
+    only_throw_errors: ignore
+
 linter:
   rules:
     - avoid_private_typedef_functions
-    - directives_ordering
-    - prefer_single_quotes
+    - prefer_expression_function_bodies
```
Comment (natebosch) [inline analysis_options.yaml:16]: @jakemac53 - do you have any concerns about enforcing this one?

File: pkgs/test_core/lib/src/bootstrap/vm.dart [hunk 2]
```diff
@@ -41,7 +41,7 @@ void internalBootstrapNativeTest(
     ..pipe(serializeSuite(getMain));
   platformChannel.sink.add(testControlChannel.id);
 
-  platformChannel.stream.forEach((message) {
+  await platformChannel.stream.forEach((message) {
```
Comment (kevmoo) [inline pkgs/test_core/lib/src/bootstrap/vm.dart:44]: unsure if this is okay...

File: pkgs/test_core/lib/src/runner/vm/platform.dart [hunk 3]
```diff
@@ -103,8 +103,8 @@ class VMPlatform extends PlatformPlugin {
       for (var fn in cleanupCallbacks) {
         fn();
       }
-      eventSub?.cancel();
-      client?.dispose();
+      await eventSub?.cancel();
```
Comment (kevmoo) [inline pkgs/test_core/lib/src/runner/vm/platform.dart:106]: unsure if this is okay...

**Review by natebosch — COMMENTED** [review]
This is a super noisy diff and it's hard to review.

Do you have handy a list of the new lints? I might try to take them one by one instead of trying to land them all at once. We'll be able to trivially approve the trivial ones, and have a much higher signal to noise to review the ones that have potential behavior changes.

File: pkgs/checks/lib/src/collection_equality.dart [hunk 4]
```diff
@@ -40,7 +40,7 @@ Iterable<String>? deepCollectionEquals(Object actual, Object expected) {
 
 const _maxDepth = 1000;
 
-class _ExceededDepthError extends Error {}
+class _ExceededDepthError implements Exception {}
```
[the remark this answers is not in the document]
Comment (kevmoo) [reply pkgs/checks/lib/src/collection_equality.dart:43]: But we CATCH it. 🤷

**Comment by kevmoo** [comment]
closing in favor of https://github.com/dart-lang/test/pull/2032

File: pkgs/checks/lib/src/collection_equality.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (natebosch) [reply pkgs/checks/lib/src/collection_equality.dart:43]: hmm, that's true... it's not really a user facing `Error` so it wouldn't matter anyway.

**Pull request closed without merging**